### PR TITLE
Makes mold a random event + spacevine oldbase port

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

This basically makes it so that a random event can occur which will spawn up to 3 molds at random locations.

Mold has also received a buff so that it's actually a threat.

Spacevines now spawn in maintenance instead of the fucking central hallways and will be spawned with two maneaters.

Also ports: 
https://github.com/Skyrat-SS13/Skyrat13/pull/3206
https://github.com/Skyrat-SS13/Skyrat13/pull/3227
https://github.com/Skyrat-SS13/Skyrat13/pull/3391

## Changelog
:cl:
add: Mold is now a random event.
balance: Spacevines have been rebalanced.
/:cl:

### CODE REVIEW

**Comment by Cheshify** [comment]
So, in my personal experience with these space vines, they grow far too rapidly, the monsters they spawn are too plentiful (15+ of them zerging into crowds of people). The damage the monsters do downs people in 1/2 hits, and since stamina damage has this insane slow attached, there's basically no way a station of people can handle the insane greentide, as well as fight off all the ridiculously-fast growing vines that have way too much health (using a full welder to clear a few randomly spawned vines?). 

Right now, this is basically a round-ending event in proportion to the Crystal Invasion, but at least everyone on station is aware of the massive yellow skeletons before they steamroll your department. This would fit well on tg's LRP servers, but not a HRP server where people aren't expected to powergame and be constantly armed to kill space vines that hit harder than a desword.

TL;DR - We need to drop the space vine's lethality by a wide margin by slowing their growth, health, and monster damage + spawnrate. Possible even locking the ability to spawn monsters behind a grown-spacevine only. If you want it to be a huge threat, give it a station announcement.

EDIT: Also vine monsters instant heal every time they walk on a vine, think it's an oversight

File: code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm [hunk 1]
```diff
@@ -130,6 +131,8 @@
 	/// Whether or not this plant is ghost possessable
 	var/playable_plant = TRUE
 
+	sight = SEE_SELF|SEE_MOBS|SEE_OBJS|SEE_TURFS // SKYRAT EDIT ADDITION
```
Comment (Seris02) [inline code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm]: why is this here when it's already there on line 121

Comment (Gandalf2k15) [inline code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm]:
```suggestion
```

File: modular_skyrat/master_files/code/modules/events/spacevine.dm [hunk 2]
```diff
@@ -0,0 +1,776 @@
...
+		if(BURN)
+			playsound(src.loc, 'sound/items/welder.ogg', 100, TRUE)
+
+/obj/structure/spacevine/Crossed(atom/movable/AM)
+	. = ..()
+	if(!isliving(AM))
+		return
+	for(var/datum/spacevine_mutation/SM in mutations)
+		SM.on_cross(src, AM)
+	if(istype(AM, /mob/living/simple_animal/hostile/venus_human_trap)) //skyrat change: vines heal flytraps 10% on cross
```
Comment (Swiftfeather) [inline modular_skyrat/master_files/code/modules/events/spacevine.dm:577]: This isn't needed, they heal when attacking living people. This just lets them fullheal then go back and fight some more, and makes them far more difficult to kill when they are already not an easy target.

File: modular_skyrat/master_files/code/modules/events/spacevine.dm [hunk 3]
```diff
@@ -0,0 +1,776 @@
...
+		if(prob(mutativeness))
+			var/datum/spacevine_mutation/randmut = pick(vine_mutations_list - SV.mutations)
+			randmut.add_mutation_to_vinepiece(SV)
+
+	for(var/datum/spacevine_mutation/SM in SV.mutations)
+		SM.on_birth(SV)
+	location.Entered(SV)
+	return SV
+
+/datum/spacevine_controller/proc/VineDestroyed(obj/structure/spacevine/S)
```
Comment (Swiftfeather) [inline modular_skyrat/master_files/code/modules/events/spacevine.dm:661]: #3355

File: modular_skyrat/master_files/code/modules/events/spacevine.dm [hunk 4]
```diff
@@ -0,0 +1,776 @@
...
+		SM.on_grow(src)
+
+/obj/structure/spacevine/proc/entangle_mob()
+	if(!has_buckled_mobs() && prob(25))
+		for(var/mob/living/V in src.loc)
+			entangle(V)
+			if(has_buckled_mobs())
+				break //only capture one mob at a time
+
+/obj/structure/spacevine/proc/entangle(mob/living/V)
```
Comment (Swiftfeather) [inline modular_skyrat/master_files/code/modules/events/spacevine.dm:723]: #3355

File: modular_skyrat/master_files/code/modules/events/spacevine.dm [hunk 5]
```diff
@@ -0,0 +1,776 @@
...
+		if(SM.process_temperature(src, exposed_temperature, volume)) //If it's ever true we're safe
+			return
+	qdel(src)
+
+/obj/structure/spacevine/CanAllowThrough(atom/movable/mover, turf/target)
+	. = ..()
+	if(isvineimmune(mover))
+		return TRUE
+
+/proc/isvineimmune(atom/A)
```
Comment (Swiftfeather) [inline modular_skyrat/master_files/code/modules/events/spacevine.dm:771]: #3355

File: modular_skyrat/master_files/code/modules/events/spacevine.dm [hunk 6]
```diff
@@ -0,0 +1,776 @@
...
+		SM.on_death(src)
+	if(master)
+		master.VineDestroyed(src)
+	mutations = list()
+	set_opacity(0)
+	if(has_buckled_mobs())
+		unbuckle_all_mobs(force=1)
+	return ..()
+
+/obj/structure/spacevine/proc/on_chem_effect(datum/reagent/R)
```
Comment (Swiftfeather) [inline modular_skyrat/master_files/code/modules/events/spacevine.dm:535]: #3355

**Comment by EverothB** [comment]
Hi, just my personal experience with this plants. So as a normal scientist with no gear but bothany gear to deal against them this was my experience. 
I reached to try and get 1 guy out of 3 plants biting someone. 1 hit me, and i was down stamcrit, unable to do anything and constantly being eaten by then once i was defenseless. 1 hit, down and dead. And i was not even fighting or with the combat hud on at full life and good stam. 
I went back trying to help get bodies, but i cant, the range on the vine toss is too big and they just hide over with the glass, we have no gear. The plants killed all sec, all the crew, all the mechs, all the borgs and all the ert. I try to use plant begone, nothing works, use a scyte, does not matter, they hit or drop you down with a vine, and its over, they gang up on you and toss you into the snow or space. How is that not abusing from meta knowledge? They are dragging the bodies away into unreachable space, and no one was able to reach them. (Not sure if this was ok, since this seem way off rping as an antag and seems clearly meta)
So plants, win, they are unkillable with all that much life steal, they instantly heal when they go into vines, and are so strong 1 hit downs you and killed all sec, any ert, people of the strongest mecs, crew just passing by, and the borgs.

File: modular_skyrat/master_files/code/modules/events/spacevine.dm [hunk 7]
```diff
@@ -0,0 +1,776 @@
...
+	for(var/mob/living/M in target)
+		if(!isvineimmune(M) && M.stat != DEAD) // Don't kill immune creatures. Dead check to prevent log spam when a corpse is trapped between vine eaters.
+			aggrospread_act(holder, M)
+
+/// What happens if an aggr spreading vine buckles a mob.
+/datum/spacevine_mutation/aggressive_spread/on_buckle(obj/structure/spacevine/holder, mob/living/buckled)
+	aggrospread_act(holder, buckled)
+
+/// Hurts mobs. To be used when a vine with aggressive spread mutation spreads into the mob's tile or buckles them.
+/datum/spacevine_mutation/aggressive_spread/aggrospread_act(obj/structure/spacevine/S, mob/living/M)
```
Comment (Swiftfeather) [inline modular_skyrat/master_files/code/modules/events/spacevine.dm:162]: #3355

**Review by Swiftfeather — CHANGES_REQUESTED** [review]
A few changes, I'm a couple drinks past beer-o-clock so I probably missed some stuff

**Comment by Gandalf2k15** [comment]
Maneaters have been rebalanced and oil mold no longer spawns plasma.

**Comment by Low3** [comment]
Is it possible to move man eater spawns and mold chance/cores amount to config? They are kinda too epic and tend to kill off rounds 1 hour in.
